### DISCUSSION TITLE
Easy contributions! Add typehint and translate docs for MMDetection~

### DISCUSSION THREAD

**Discussion by hhaAndroid** [open-mmlab/mmdetection#9234] [General]
Hi guys, we are planning to add typehint and docstring for python code and translate docs from English to Chinese in MMDetection. PRs from the community are welcomed. To make the community fully aware of the progress, we list the progress here. Please feel free to leave a message and create a PR if you are willing to do these.

- [ ] part-1 Owner @simpleiswell
     - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/datasets/api_wrappers/coco_api.py 
     - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/datasets/utils.py 
     - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/engine/hooks/set_epoch_info_hook.py 
     - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/engine/hooks/sync_norm_hook.py
     - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/engine/hooks/utils.py 
     - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/engine/optimizers/layer_decay_optimizer_constructor.py 
     - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/evaluation/functional/bbox_overlaps.py 
- [ ] part-2 Owner @cyhrane
    - [ ]  https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/evaluation/functional/mean_ap.py 
    - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/evaluation/functional/panoptic_utils.py 
    - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/evaluation/functional/recall.py 
- [ ] part-3 Owner @hjy94wo777
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/csp_darknet.py #9402
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/darknet.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/detectors_resnet.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/detectors_resnext.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/efficientnet.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/hrnet.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/mobilenet_v2.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/pvt.py 
- [ ] part-4 Owner @pgh79
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/regnet.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/res2net.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/resnest.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/resnet.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/resnext.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/ssd_vgg.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/swin.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/backbones/trident_resnet.py 
- [ ] part-5
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/layers/brick_wrappers.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/layers/conv_upsample.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/layers/csp_layer.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/layers/dropblock.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/layers/inverted_residual.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/layers/matrix_nms.py 
- [x] part-6 Owner @NoFish-528 #9346
   - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/layers/normed_predictor.py 
   - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/layers/positional_encoding.py 
   - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/layers/res_layer.py 
   - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/layers/se_layer.py 
- [ ] part-7 Owner @SmallingCar
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/accuracy.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/ae_loss.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/balanced_l1_loss.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/cross_entropy_loss.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/dice_loss.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/focal_loss.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/gfocal_loss.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/ghm_loss.py 
- [x] part-8 Owner @lihua199710  https://github.com/open-mmlab/mmdetection/pull/9397
   - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/iou_loss.py 
   - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/kd_loss.py 
   - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/mse_loss.py 
   - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/seesaw_loss.py 
   - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/smooth_l1_loss.py 
   - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/utils.py 
   - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/losses/varifocal_loss.py 
- [x] part-9 Owner @JunyaoHu
   - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/necks/bfp.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/necks/channel_mapper.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/necks/dilated_encoder.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/necks/dyhead.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/necks/fpg.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/necks/fpn_carafe.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/necks/hrfpn.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/necks/nasfcos_fpn.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/necks/pafpn.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/necks/rfp.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/necks/ssd_neck.py
   - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/necks/yolo_neck.py #9383
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/necks/yolox_pafpn.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/roi_heads/shared_heads/res_layer.py
- [ ] part-10
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/coders/bucketing_bbox_coder.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/coders/delta_xywh_bbox_coder.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/coders/distance_point_bbox_coder.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/coders/legacy_delta_xywh_bbox_coder.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/coders/pseudo_bbox_coder.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/coders/tblr_bbox_coder.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/coders/yolo_bbox_coder.py 
- [x] part-11 Owner @QJC123654 #9468
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/prior_generators/anchor_generator.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/prior_generators/point_generator.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/prior_generators/utils.py 
- [ ] part-12
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/samplers/combined_sampler.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/samplers/instance_balanced_pos_sampler.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/samplers/iou_balanced_neg_sampler.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/samplers/ohem_sampler.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/samplers/sampling_result.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/builder.py
- [ ] part-13 Owner @naixie
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/utils/gaussian_target.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/utils/make_divisible.py 
- [ ] part-14
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/utils/misc.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/builder.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/utils/compat_config.py
- [ ] part-15 @fuyueagain
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/structures/bbox/bbox_overlaps.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/structures/mask/mask_target.py
- [ ] part-16
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/structures/mask/structures.py 
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/structures/mask/utils.py 
- [ ] part-17
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/utils/contextmanagers.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/utils/dist_utils.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/utils/logger.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/utils/memory.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/utils/misc.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/utils/profiling.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/utils/split_batch.py
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/version.py

Docs translation or update

- [ ] part-1 Owner @137208  https://github.com/open-mmlab/mmdetection/pull/9434
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/docs/zh_cn/user_guides/single_stage_as_rpn.md
   - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/docs/zh_cn/user_guides/test_results_submission.md
- [x] part-2 Owner @tianleiSHI  https://github.com/open-mmlab/mmdetection/pull/9453
    - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/docs/zh_cn/user_guides/useful_tools.md
    - [x] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/docs/zh_cn/user_guides/useful_hooks.md
- [ ] part-3 Owner @Laymanpython
    - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/docs/zh_cn/user_guides/test.md
    - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/docs/zh_cn/user_guides/train.md
- [ ] part-4 Owner @DanShouzhu https://github.com/open-mmlab/mmdetection/pull/9578
    - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/docs/zh_cn/user_guides/finetune.md
    - [ ] https://github.com/open-mmlab/mmdetection/blob/dev-3.x/docs/zh_cn/user_guides/inference.md

**Comment by ZhikangNiu** [comment]
牛志康 type hints part6

**Comment by hjy94wo777** [comment]
胡健谕 type hints part-3

**Comment by naixie** [comment]
林钰睿 type hints part-13

**Comment by pgh79** [comment]
彭国宏 type hints part-4

**Comment by QJC123654** [comment]
祁军成 type hints part-11

**Comment by JunyaoHu** [comment]
胡钧耀 type hints part-9

**Comment by SmallingCar** [comment]
车艳波 type hints part-7

**Comment by Laymanpython** [comment]
李鑫 Docs translation or update part-3

**Comment by ZXH-NEU** [comment]
张新慧 type hints part1

**Comment by lihua199710** [comment]
liteng type hints part-8

**Comment by SmallingCar** [comment]
I will try it. Tank you very much.

在 2022-11-22 20:46:16，Laymanpython ***@***.***> 写道：


李鑫 Docs translation or update part-3

**Comment by Laymanpython** [comment]
李鑫 Docs translation or update part-3

---Original---
From: ***@***.***&gt;
Date: Sun, Nov 27, 2022 04:23 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [open-mmlab/mmdetection] Easy contributions! Add typehint and translate docs for MMDetection~ (Discussion #9234)




 I will try it. Tank you very much.
 
 在 2022-11-22 20:46:16，Laymanpython ***@***.***&gt; 写道：
 
 
 李鑫 Docs translation or update part-3

**Comment by tianleiSHI** [comment]
石田蕾  Docs translation or update part-2

**Comment by cyhrane** [comment]
邱奕涵 type hints part2

**Comment by 137208** [comment]
刘浩 Docs translation or update part1

**Comment by fuyueagain** [comment]
wang yingying type hints part15

**Comment by DanShouzhu** [comment]
淡寿祝 Docs translation or update part 4

**Comment by IRONICBo** [comment]
Asklv type hints part-10

**Reply by IRONICBo** [reply]
https://github.com/open-mmlab/mmdetection/pull/10040
